source/docs/advanced/payments/custom-payment-information.md

```diff
@@ -0,0 +1,138 @@
+
+## Display a custom input for a payment method
+
+You should register a new component in `theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js` by following this template:
+
+```js
+import CustomComponent from "theme/modules/CustomComponent/CustomComponent";
+
+const ComponentMap = {
+  "<method_code>": CustomComponent,
+};
+
+// ... the rest of the code should be kept intact
+```
+
+The Custom component referenced here should be created in your own module, and display the additional information of your payment method to the user.
+
+```js
+// theme/modules/CustomComponent/CustomComponent.js
+import React from "react";
+import SubmitPayment from "theme/modules/Checkout/Payment/SubmitPayment";
+
+const CustomComponent = ({
+  onAuthorize,
+  value,
+  gscAccepted,
+  error,
+  method,
+}) => {
+  const [comments, setComments] = useState(value?.comments ?? "");
+
+  return (
+    <div className="custom-component">
+      <input
+        onChange={(event) => setComments(event.target.value)}
+        value={comments}
+        type="text"
+      />
+      <SubmitPayment
+        gscAccepted={gscAccepted}
+        onSubmit={() => {
+          onAuthorize(gscAccepted ? { comments } : null);
+        }}
+        error={this.props.error}
+      />
+    </div>
+  );
+};
+
+export default CustomComponent;
+```
+
+Things to understand here:
+
+- The `SubmitPayment` component is a common component across all payment methods to ensure consistency in how methods are managed.
+- `method` the current selected method.
+- `value` contains initial additional data (usually it is always null).
+- `onAuthorize` should be called with the additional data on `<SubmitPayment>`'s on submit method.
+- `gscAccepted` whether or not the user have accepted the general sales conditions (checkbox above the "Place my order" button). Just forward this to the `<SubmitPayment>` component.
+- `error` an error object if there is an error. Just forward this to the `<SubmitPayment>` component.
+
+In this example we're only setting a comments additional text input, but you can do fancier things like fetching a list for the user to pick from GraphQL and displaying them here.
+
+## Updating an existing method Additional Data
+
+Sometimes you may need to add an extra field to additional data of an existing payment method. To do so you need to first find if this payment method already has an `AdditionalDataComponent` registered. In case it does not have an `AdditionalDataComponent` registered what was in [Display a custom input for a payment method](#Display-a-custom-input-for-a-payment-method) will work fine. On the other hand if it has `AdditionalDataComponent` defined you can override the `AdditionalDataComponent` for that method and add your custom modifications there.
```

Comment:
I'm not sure how checkboxes format in our docs, so just check the formatting first locally

```suggestion
Sometimes you may need to add an extra field to the additional data of an existing payment method. You first need to check if this payment method already has an `AdditionalDataComponent` registered.
- [x] If `AdditionalDataComponent` has been registered, you should then override the `AdditionalDataComponent` and add your custom modifications.
- [ ] If no `AdditionalDataComponent` is registered, you can then follow the [Display a custom input for a payment method](#Display-a-custom-input-for-a-payment-method) docs.
```